kafka-event-listener/src/main/java/org/hypertrace/core/kafka/event/listener/KafkaLiveEventListener.java

```diff
@@ -101,7 +101,7 @@ class ListenerThreadFactory implements ThreadFactory {
   private final String name;
 
   public ListenerThreadFactory(String consumerName) {
-    this.name = "kafka-live-event-listener-" + consumerName;
+    this.name = "kafka-live-event-listener-" + consumerName + "-" + System.currentTimeMillis();
```

Comment:
Is it common pattern to add time in name?

Reply:
I added here intentionally to detect if we have multiple threads for same consumer name

Reply:
Ideally for one consumer there should be only one live event listener hence one thread only. But if it is wrongly used there might be more than one, which we want to detect and fix

Reply:
> If, its more than that, who is caller? How would we fix this? I think, is this called by our application code or by kafka framework itself?

This is used by application code, mainly to listen to change events. If we see more than one thread for a consumer name that means we are rebuilding live event listener under some condition, which is wrong. To fix, you might need to organise the building of live event listener to the place where code is exactly once executed - guice singleton or build topology and pass it down appropriately